src/lib/operators/table_scan.cpp

```diff
@@ -30,59 +39,67 @@
 
 namespace opossum {
 
-TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const OperatorScanPredicate& predicate)
-    : AbstractReadOnlyOperator{OperatorType::TableScan, in}, _predicate{predicate} {}
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in,
+                     const std::shared_ptr<AbstractExpression>& predicate)
+    : AbstractReadOnlyOperator{OperatorType::TableScan, in}, _predicate(predicate) {}
 
 TableScan::~TableScan() = default;
 
 void TableScan::set_excluded_chunk_ids(const std::vector<ChunkID>& chunk_ids) { _excluded_chunk_ids = chunk_ids; }
 
+const std::shared_ptr<AbstractExpression>& TableScan::predicate() const { return _predicate; }
+
 const std::string TableScan::name() const { return "TableScan"; }
 
 const std::string TableScan::description(DescriptionMode description_mode) const {
   const auto separator = description_mode == DescriptionMode::MultiLine ? "\n" : " ";
-  return name() + separator + _predicate.to_string(input_table_left());
-}
 
-const OperatorScanPredicate& TableScan::predicate() const { return _predicate; }
+  std::stringstream stream;
 
-void TableScan::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {
-  if (is_parameter_id(_predicate.value)) {
-    const auto value_iter = parameters.find(boost::get<ParameterID>(_predicate.value));
-    if (value_iter != parameters.end()) _predicate.value = value_iter->second;
+  stream << name() << separator;
+  if (_impl) {
+    stream << "Impl: " << _impl->description();
+  } else {
+    stream << "Impl unset";
```

Comment:
This is what I get when I try to visualize the TPC-H plans. :/

Reply:
via the console? haven't looked at that yet
via the benchmarks, in subqueries?